camelot/core/serializable.py

```diff
@@ -88,6 +88,8 @@ def asdict(cls, obj):
     def _asdict_inner(cls, obj):
         if dataclasses._is_dataclass_instance(obj):
             return cls.serialize_fields(obj)
+        elif isinstance(obj, Serializable):
+            return obj._to_dict()
```

Comment:
hmm, _to_dict is bedoeld om in unittests te gebruiken, de implementatie is nogal inefficient ... 
mss moeten we van documents wel een dataclass maken of de asdict method overschrijven voor de action step die documents bevat ... tbd